src/main/java/io/kestra/plugin/kafka/RealtimeTrigger.java

```diff
@@ -94,7 +115,69 @@ public Publisher<Execution> evaluate(ConditionContext conditionContext, TriggerC
             .maxDuration(this.maxDuration)
             .build();
 
-        return Flux.from(task.stream(runContext))
+        return Flux.from(publisher(task, runContext))
             .map((record) -> TriggerService.generateRealtimeExecution(this, context, task.recordToMessage(record)));
     }
+
+    public Publisher<ConsumerRecord<Object, Object>> publisher(final Consume task,
+                                                               final RunContext runContext) {
+        return Flux.create(fluxSink -> {
+            try (KafkaConsumer<Object, Object> consumer = task.consumer(runContext)) {
+                this.consumer.set(consumer);
+                task.topicSubscription(runContext).subscribe(consumer, task);
+                while (isActive.get()) {
+                    try {
+                        consumer.poll(task.getPollDuration()).forEach(fluxSink::next);
+                        consumer.commitSync();
+                    } catch (org.apache.kafka.common.errors.InterruptException e) {
+                        // ignore, this case is handle by next lines
+                    }
+                    // Check if the current thread has been interrupted before next poll.
+                    if (Thread.currentThread().isInterrupted()) {
+                        isActive.set(false); // proactively stop polling
+                    }
+                }
+            } catch (WakeupException e) {
+                // ignore and stop
+            } catch (Throwable e) {
+                fluxSink.error(e);
+            } finally {
+                fluxSink.complete();
+                this.waitForTermination.countDown();
+            }
+        }, FluxSink.OverflowStrategy.BUFFER);
```

Comment:
```suggestion
        });
```

The overlow strategy is choosed by the worker